### DISCUSSION TITLE
Removing default header

### DISCUSSION THREAD

**Discussion by ghost** [axios/axios#5904] [Q&A]
Hi, what's the difference between these two:
`axios.defaults.headers["User-Agent"] = null;`
`axios.defaults.headers.common["User-Agent"] = null;`
Both seem to work

**Comment by gitsult4n** [comment]
`axios.defaults.headers.common` is the documented bucket: it's applied to every method, and can be overridden by the method-specific buckets (`headers.get`, `headers.post`, etc.).

Top-level `axios.defaults.headers["User-Agent"]` sits outside that bucket system. When a request is built, axios merges `headers.common` with `headers[method]`, strips the method buckets out, then concatenates the remaining top-level keys on top — so a top-level assignment ends up taking precedence over both `common` and the method-specific buckets.

Both appear to work in your case simply because nothing else is setting that header. Prefer `common` — it's the intended API; the top-level form depends on a merge-order implementation detail rather than a documented contract.
